src/routes/api/decks/shared/+server.ts

```diff
@@ -1,31 +1,39 @@
-import { authenticate, getUserFromEmail, pushPath, readPath, writePath } from '$lib/server/firebaseUtils';
+import { authenticate, getUserFromEmail, pushPath, readPath, getUserData } from '$lib/server/firebaseUtils';
 import { error, json } from '@sveltejs/kit';
 import type { RequestHandler } from './$types';
 import { getStripeCustomerWithSubscriptions, isSubscribedToBlendPro } from '$lib/server/subscriptionUtils';
+import { sendDeckShareEmail } from '$lib/server/emailUtils';
 
 export const POST = (async (event) => {
   const { uid: sourceUid } = await authenticate(event);
   const { deckId, targetUserEmail } = await event.request.json();
-  if (!targetUserEmail || !deckId) 
+  if (!targetUserEmail || !deckId)
     throw error(400, "No target email and/or deckId provided");
-  const { uid: targetUid } = await getUserFromEmail(targetUserEmail);
-  const [sourceCustomer, targetCustomer, deck] = await Promise.all([
+  const targetUserPromise = getUserFromEmail(targetUserEmail).then((userData) => Promise.all([userData, getStripeCustomerWithSubscriptions(userData.uid)]));
+  const [
+    { displayName: sourceUserName = 'A Blend user' },
+    [{ displayName: targetUserName }, targetCustomer],
+    sourceCustomer,
+    deck
+  ] = await Promise.all([
+    getUserData(sourceUid),
+    targetUserPromise,
     getStripeCustomerWithSubscriptions(sourceUid),
-    getStripeCustomerWithSubscriptions(targetUid),
     readPath(`/decks/user/${sourceUid}/${deckId}`)
   ]);
 
-  if (!deck) 
+  if (!deck)
     throw error(404, `Deck ${deckId} does not exist for user ${sourceUid}`);
   if (!isSubscribedToBlendPro(sourceCustomer))
     throw error(401);
   if (!isSubscribedToBlendPro(targetCustomer))
     throw error(400, `The user ${targetUserEmail} is not a Blend Pro subscriber`);
 
-  const sharedKey = (await pushPath('/decks/shared', { sourceUid, createdAt: Date.now(), deck })).key;
-  return json({ sharedKey }, { status: 201, headers: [ ['Access-Control-Allow-Origin', "*"] ] })
+  const sharedKey = (await pushPath('/decks/shared', { sourceUid, createdAt: Date.now(), deck })).key!;
+  await sendDeckShareEmail(targetUserEmail, targetUserName, { deckName: deck.name, shareId: sharedKey, sender: sourceUserName })
```

Comment:
We could return a little faster by not waiting for this but I figured if the email fails to send then we should fail it